src/Form/Form.jsx

```diff
@@ -0,0 +1,191 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import JSONSchemaForm from 'react-jsonschema-form';
+import {ProgressBar, Intent} from '@blueprintjs/core';
+
+import isEqual from 'lodash/isEqual';
+import merge from 'lodash/merge';
+
+import {
+  getUiSchemaLogic,
+  getUiSchemaProperties,
+  getUiSchemaTitle
+} from '../uiSchema/UiSchemaSelectors';
+
+import fields from './formComponents/fields';
+import widgets from './formComponents/widgets';
+import Template from './formComponents/Template';
+import ErrorListTemplate from './formComponents/ErrorListTemplate';
+import {
+  getLoadingState,
+  getSchema,
+  getError
+} from './FormSelectors';
+import {
+  prepareSchema,
+  clearData,
+  clearError
+} from './FormActions';
+import {removeEmpty, toServerData, prepareFormData} from './utils';
+
+/**
+ * Form component for creating and editing resources
+ *
+ * @class Form
+ */
+export class Form extends Component {
+
+  /**
+   * Reference to React JSON Schema Form component
+   * @type {ReactElement}
+   */
+  form = null;
+
+  /**
+   * Prepares schema to before shows dialog.
+   * @override
+   */
+  componentDidMount() {
+    const {baseSchema, action, formName} = this.props;
+
+    // temporary solution
+    this.props.onRef(this);
+
+    this.props.prepareSchema(formName, baseSchema.schema, action, baseSchema.parent, {
+      ...merge(this.props.jsonUiSchema, this.props.uiSchema)
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {action, formName} = this.props;
+
+    if (!isEqual(this.props.baseSchema.schema, nextProps.baseSchema.schema)) {
+      this.props.prepareSchema(formName, nextProps.baseSchema.schema, action, nextProps.baseSchema.parent, {
+        ...merge(this.props.jsonUiSchema, this.props.uiSchema)
+      });
+    }
+  }
+
+  /**
+   * Clears all data from store.
+   * @override
+   */
+  componentWillUnmount() {
+    this.props.onRef(null);
+    this.props.clearData(this.props.formName);
```

Comment:
Please call there also `clearError`. Now when we get error in one dialog and then close it and open another one we get the same error because store is not cleared.

Reply:
Done